### PR TITLE
[2.10] Fix Tree.seq which creates unnecessary pruned hashes after #1683

### PR DESCRIPTION
**Pull request by Ngoguey42**
I was a bit hasty when I introduced changes on `Tree.seq` in #1683, which was necessary for merkle proofs.

As the code exists in the 2.10 branch, `seq` may produce unnecessary pruned hashes.

I'm a bit surprised that the problem did not occur yet.

### CODE REVIEW

**Comment by samoht** [comment]
Could you add a regression test as well?

**Comment by clecat** [comment]
Thank you a lot for this fix

**Comment by Ngoguey42** [comment]
> Could you add a regression test as well?

Done

**Review by icristescu — APPROVED** [review]
That was tricky, good catch! 

I think you need to update the opam file for the CI to pass (as [here](https://github.com/mirage/irmin/pull/1710/commits/9dec7a292006bce2331f30a5a15fc70f9aff27a5#diff-b5c4e36278a4cc327559c1646f6277074a7457a24a268305ec4d4751b7f7cf90))

File: test/irmin/test_tree.ml [hunk 1]
```diff
@@ -85,7 +85,20 @@ let test_bindings _ () =
       "Bindings are reported in lexicographic order" [ "a"; "aa"; "ab"; "b" ]
   in
   (* [Tree.list] returns all keys in lexicographic order *)
-  Tree.list tree [] >|= (List.map fst >> check_sorted)
+  let* () = Tree.list tree [] >|= (List.map fst >> check_sorted) in
+
+  (* Make sure that a [Hash (Some repo, _)] node with a cached value doesn't
+     produce a list of pruned children. See #1720 *)
```
Comment (icristescu) [inline test/irmin/test_tree.ml:91]:
```suggestion
     produce a list of pruned children. *)
```
I think there is no need to link to the PR, as there are no more info in the PR than in the test, no?

**Pull request merged**